src/main/java/org/opensearch/ad/rest/RestValidateDetectorAction.java

```diff
@@ -1,44 +1,122 @@
 package org.opensearch.ad.rest;
 
+import static org.opensearch.ad.util.RestHandlerUtils.TYPE;
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
 import static org.opensearch.rest.RestRequest.Method.POST;
-import static org.opensearch.rest.RestStatus.NOT_FOUND;
-import static org.opensearch.rest.RestStatus.OK;
 
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
+import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.opensearch.ad.AnomalyDetectorExtension;
+import org.opensearch.ad.common.exception.ADValidationException;
 import org.opensearch.ad.constant.CommonErrorMessages;
+import org.opensearch.ad.model.AnomalyDetector;
+import org.opensearch.ad.model.DetectorValidationIssue;
+import org.opensearch.ad.model.ValidationAspect;
 import org.opensearch.ad.settings.EnabledSetting;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorRequest;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentType;
 import org.opensearch.extensions.rest.ExtensionRestRequest;
 import org.opensearch.extensions.rest.ExtensionRestResponse;
-import org.opensearch.rest.RestHandler.Route;
-import org.opensearch.rest.RestRequest.Method;
-import org.opensearch.sdk.ExtensionRestHandler;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.sdk.BaseExtensionRestHandler;
+import org.opensearch.sdk.ExtensionsRunner;
+import org.opensearch.sdk.RouteHandler;
 
-public class RestValidateDetectorAction implements ExtensionRestHandler {
+public class RestValidateDetectorAction extends BaseExtensionRestHandler {
     private final Logger logger = LogManager.getLogger(RestValidateDetectorAction.class);
+    private final OpenSearchClient sdkClient;
+    private final NamedXContentRegistry xContentRegistry;
+
+    public static final Set<String> ALL_VALIDATION_ASPECTS_STRS = Arrays
+        .asList(ValidationAspect.values())
+        .stream()
+        .map(aspect -> aspect.getName())
+        .collect(Collectors.toSet());
+
+    public RestValidateDetectorAction(ExtensionsRunner runner, AnomalyDetectorExtension extension) {
+        this.xContentRegistry = runner.getNamedXContentRegistry().getRegistry();
+        this.sdkClient = extension.getClient();
+    }
 
     @Override
-    public List<Route> routes() {
-        return List.of(new Route(POST, "/detectors/_validate"), new Route(POST, "/detectors/_validate/{type}"));
+    protected List<RouteHandler> routeHandlers() {
+        return List.of(new RouteHandler(POST, "/detectors/_validate", (r) -> handleRequest(r)));
```

Comment:
The `handleRequest()` method below is an override of the superclass method which actually does the filtering/parsing of the route handler.  So in reality you'll never actually use the `(r) -> handleRequest(r)` part here; it'll jump straight to the overridden method and ignore all the snazzy route handler method-and-path-checking goodness.

While that happens to give us the result we want in this case, it's not clear.  This is "shadowing" and is almost always a Bad™️ thing.

So pick any other name for the request handler below, so that it doesn't override.

Reply:
ACK